astropy/cosmology/flrw.py

```diff
@@ -1805,7 +1807,7 @@ def _hypergeometric_comoving_distance_z1z2(self, z1, z2):
         except ValueError as e:
             raise ValueError("z1 and z2 have different shapes") from e
 
-        s = ((1 - self._Om0) / self._Om0) ** (1./3)
+        s = power((1 - self._Om0) / self._Om0, 1.0/3)
```

Comment:
Is this value ever negative? Or more specifically is imaginary `s` meaningful? If it is, I think it deserves some test cases that the behaviour is in fact correct.

Reply:
I actually had to use this ``power`` function because ``s`` is negative at some point in the test suite and was failing the CI.

Reply:
But what does it mean physically to have imaginary `s`?

Reply:
Tracked down why the CI started failing. It's because doing ``Quantity(x).value`` makes a ``float`` into a ``numpy.float64``. These do powers differently 

```python
>>> ((1 - 3.0) / 3.0) ** (1 / 3)
(0.43679023236814957+0.7565428747114508j)

>>> ((1 - np.float64(3.0)) / np.float64(3.0)) ** (1 / 3)
/var/folders/....py:1: RuntimeWarning: invalid value encountered in double_scalars
  ((1 - np.float64(3.0)) / np.float64(3.0)) ** (1 / 3)
[15]: nan

>>> from numpy.lib.scimath import power
>>> power((1 - np.float64(3.0)) / np.float64(3.0), (1 / 3))
(0.43679023236814957+0.7565428747114508j)
```

So the ``power`` is necessary not because I've changed the math, but because I've changed the ``type`` of the input.
As to whether  the existing code is correct and ``s`` should ever be imaginary... I think only for closed universes. I'll add that to my Project Board to follow up on.

Reply:
Added to project

Reply:
Perhaps I should change the Parameter ``dtype=`` default from ``np.float64`` back to ``float``.
For the Array Cosmology project, the goal is use NumPy types for the Parameters, but maybe I'm jumping the gun in making the change now.

Reply:
Actually ``Quantity(x).value`` changes ``float`` to ``float64``, so this change is inevitable.